### DISCUSSION TITLE
does sherpa can work in production for streaming?

### DISCUSSION THREAD

**Discussion by trangtv57** [k2-fsa/sherpa#88] [Ideas]
I find the solution asr end2end can work in production for streaming. I don't know sherpa can work in production or not? And if It's not, What I need to do to make sherpa can work in production.
Tks you

**Comment by csukuangfj** [comment]
What are your requirements?

We have streaming ASR examples:
https://k2-fsa.github.io/sherpa/streaming_asr/index.html

**Reply by trangtv57** [reply]
tks @csukuangfj 
I want sherpa can work like kaldi for streaming, can serve many request in same time with low RTF. I am not deep dive to understand how sherpa work in streaming, but I understand that can be streaming and streaming in production may be has big gap?. 
Tks

**Reply by csukuangfj** [reply]
Now sherpa can support both streaming and non-streaming ASR. It also provides websocket server/client, both in C++ and Python.
There are also C++ APIs that you can use in your project.
